From b84682f31dc99b9c90f5a04947075815697c68d9 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jose.dapena@lge.com>
Date: Wed, 7 Mar 2018 20:07:48 +0000
Subject: [PATCH] GCC: do not initialize NEON int32x4_t with braces initializer

GCC does not support int32x4_t loading using braces wrapping 4 ints.
Instead we should use the NEON intrinsic to load from an array of
ints.

Bug: 819294
Change-Id: I13b877405273e4ebcc944d50c155ee29ff31cc99
Reviewed-on: https://chromium-review.googlesource.com/951773
Commit-Queue: Mike Klein <mtklein@chromium.org>
Reviewed-by: Mike Klein <mtklein@chromium.org>
Cr-Commit-Position: refs/heads/master@{#541544}
---
 skia/ext/convolver_neon.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/skia/ext/convolver_neon.cc b/skia/ext/convolver_neon.cc
index 26b91b9c88c4..cae6bc2f833f 100644
--- a/skia/ext/convolver_neon.cc
+++ b/skia/ext/convolver_neon.cc
@@ -23,7 +23,7 @@ AccumRemainder(const unsigned char* pixels_left,
     remainder[2] += coeff * pixels_left[i * 4 + 2];
     remainder[3] += coeff * pixels_left[i * 4 + 3];
   }
-  return {remainder[0], remainder[1], remainder[2], remainder[3]};
+  return vld1q_s32(remainder);
 }
 
 // Convolves horizontally along a single row. The row data is given in
@@ -336,4 +336,4 @@ void ConvolveVertically_Neon(const ConvolutionFilter1D::Fixed* filter_values,
   }
 }
 
-}  // namespace skia
\ No newline at end of file
+}  // namespace skia
diff --git a/skia/BUILD.gn b/skia/BUILD.gn
index 79aafb2b535f..a8daf9b77911 100644
--- a/skia/BUILD.gn
+++ b/skia/BUILD.gn
@@ -282,10 +282,12 @@ component("skia") {
       "ext/convolver_mips_dspr2.h",
     ]
   } else if (current_cpu == "arm" || current_cpu == "arm64") {
-    sources += [
-      "ext/convolver_neon.cc",
-      "ext/convolver_neon.h",
-    ]
+    if (arm_use_neon) {
+      sources += [
+        "ext/convolver_neon.cc",
+        "ext/convolver_neon.h",
+      ]
+    }
   }
 
   # The imported Skia gni source paths are made absolute by gn.
